docs/advanced/custom-map-controls.md

```diff
@@ -0,0 +1,66 @@
+# Custom Map Controls
+
+## Overriding The Default Map Controller
+
+To change the default behavior of map interaction, you can implement/extend the `MapControls`
```

Comment:
Might want to be explicit that any custom implementation (that does not directly extend MapControls) must implement the API as documented below.